The job **`pr-check-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16709157338/job/47291523179) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47291523179) [(plain)](https://github.com/rust-lang/rust/commit/0fc453f0f69da91d1ca5a6455247cd1ca38f903b/checks/47291523179/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] cargo_clippy test:false 0.121
error[E0658]: use of unstable library feature `path_file_prefix`
  --> src/tools/clippy/tests/missing-test-files.rs:31:24
   |
31 |         match x.path().file_prefix().cmp(&y.path().file_prefix()) {
   |                        ^^^^^^^^^^^
   |
   = note: see issue #86319 <https://github.com/rust-lang/rust/issues/86319> for more information
   = help: add `#![feature(path_file_prefix)]` to the crate attributes to enable
   = note: this compiler was built on 2025-06-23; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `path_file_prefix`
  --> src/tools/clippy/tests/missing-test-files.rs:31:52
   |
31 |         match x.path().file_prefix().cmp(&y.path().file_prefix()) {
   |                                                    ^^^^^^^^^^^
   |
   = note: see issue #86319 <https://github.com/rust-lang/rust/issues/86319> for more information
   = help: add `#![feature(path_file_prefix)]` to the crate attributes to enable
   = note: this compiler was built on 2025-06-23; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `path_file_prefix`
  --> src/tools/clippy/tests/missing-test-files.rs:52:36
   |
52 |             let file_prefix = path.file_prefix().unwrap().to_str().unwrap().to_string();
   |                                    ^^^^^^^^^^^
   |
   = note: see issue #86319 <https://github.com/rust-lang/rust/issues/86319> for more information
   = help: add `#![feature(path_file_prefix)]` to the crate attributes to enable
   = note: this compiler was built on 2025-06-23; consider upgrading it if it is out of date
```

</details>

--- a/library/std/src/path.rs
+++ b/library/std/src/path.rs
@@ -2678,7 +2678,6 @@ impl Path {
     /// # Examples
     ///
     /// ```
-    /// # #![feature(path_file_prefix)]
     /// use std::path::Path;
     ///
     /// assert_eq!("foo", Path::new("foo.rs").file_prefix().unwrap());
@@ -2691,7 +2690,7 @@ impl Path {
     ///
     /// [`Path::file_stem`]: Path::file_stem
     ///
-    #[unstable(feature = "path_file_prefix", issue = "86319")]
+    #[stable(feature = "path_file_prefix", since = "CURRENT_RUSTC_VERSION")]
     #[must_use]
     pub fn file_prefix(&self) -> Option<&OsStr> {
         self.file_name().map(split_file_at_dot).and_then(|(before, _after)| Some(before))

--- a/library/std/tests/path.rs
+++ b/library/std/tests/path.rs
@@ -1,10 +1,4 @@
-#![feature(
-    clone_to_uninit,
-    path_add_extension,
-    path_file_prefix,
-    maybe_uninit_slice,
-    normalize_lexically
-)]
+#![feature(clone_to_uninit, path_add_extension, maybe_uninit_slice, normalize_lexically)]
 
 use std::clone::CloneToUninit;
 use std::ffi::OsStr;

--- a/src/tools/clippy/tests/missing-test-files.rs
+++ b/src/tools/clippy/tests/missing-test-files.rs
@@ -1,6 +1,6 @@
 #![warn(rust_2018_idioms, unused_lifetimes)]
 #![allow(clippy::assertions_on_constants)]
-#![feature(path_file_prefix)]
+#![cfg_attr(bootstrap, feature(path_file_prefix))]
 
 use std::cmp::Ordering;
 use std::ffi::OsStr;

